File: libs/document-processor/src/repo/core/schema/DocumentUpdateChannel.ts
import { z } from 'zod';
import { blueIds } from '../blue-ids';
import { withTypeBlueId } from '@blue-labs/language';
import { ChannelSchema } from './Channel';

export const DocumentUpdateChannelSchema = withTypeBlueId(
  blueIds['Document Update Channel']
)(
  ChannelSchema.extend({
    path: z.string().optional(),
  })
);

export type DocumentUpdateChannel = z.infer<typeof DocumentUpdateChannelSchema>;

File: references/java-sdk/src/main/java/blue/language/processor/model/DocumentUpdateChannel.java
package blue.language.processor.model;

import blue.language.model.TypeBlueId;
import blue.language.types.TypeAlias;

@TypeAlias("Document Update Channel")
@TypeBlueId({
        "6H1iGrDAcqtFE1qv3iyMTj79jCZsMUMxsNUzqYSJNbyR",
        "Document Update Channel",
        "Core/Document Update Channel",
        "DocumentUpdateChannel"
})
public class DocumentUpdateChannel extends ChannelContract {

    private String path;

    public String getPath() {
        return path;
    }

    public DocumentUpdateChannel setPath(String path) {
        this.path = path;
        return this;
    }

    public DocumentUpdateChannel path(String path) {
        return setPath(path);
    }
}
